### ISSUE TITLE
Infinite loop after evaluation " 'x' => -1" within match() 

### ISSUE THREAD

**Issue by BromStyle** [squizlabs/PHP_CodeSniffer#3679]
**Describe the bug**
phpcs / cif runs into an infinity loop alternating between two error messages:

phpcs: Expected at least 1 space after "-"; 0 found
in case of "0 => -1"

and
phpcs: Expected exactly 0 space after "-", 1 found.
in case of "0 => - 1"

**Code sample**
```php
        $rand       = rand(0, 10);
        $correction = match ($rand) {
            0       => -1,
            9, 10   => 1,
            default => 0
        };
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------
 7 | ERROR | [x] Expected exactly 0 space after "-", 1 found.
   |       |     (SlevomatCodingStandard.Operators.NegationOperatorSpacing.InvalidSpaceAfterMinus)
------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY

```

**Expected behavior**
expecting either space or no space after "-" but not both^^

**Versions (please complete the following information):**
 - OS: MacOS 11.15
 - PHP: 8.0
 - PHPCS: 3.7.1 (stable)

**Comment by jrfnl** [comment]
Please report this to the [Slevomat Coding Standard](https://github.com/slevomat/coding-standard/) project. Nothing we can do about it here.

**Comment by jrfnl** [comment]
Note: this may also be something to do with your ruleset containing contradictory rules, but that can't be evaluated without more information.
